DAOS-2153 debug: Update to gurt debug log version 2

jolivier23 opened the pull request:
This is an optimization when logging is disabled as it
bypasses the bitwise arithmetic and several checks,
replacing with a single check for all but the first
check.
Also, use daos_ prefix for log variables
Updates to latest cart to get the feature
Updates scons_local to latest to use github.com

Change-Id: I8940de203e0911a2ab097a6566a6624514417a67
Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

daosbuild1 commented on src/include/daos/debug.h:
```diff
@@ -26,39 +26,60 @@
-extern int d_dfs_logfac;
-extern int d_drpc_logfac;
-extern int d_security_logfac;
+#define DAOS_FOREACH_LOG_FAC(ACTION, arg)	\
```
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 commented on src/include/daos/debug.h:
```diff
@@ -26,39 +26,60 @@
+	ACTION(security,  security,  arg)
 
-#include <gurt/debug.h>
+#define DAOS_FOREACH_DB(ACTION, arg)				\
```
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-175/1/display/redirect

daosbuild1 commented:
Test stage Functional completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-175/1/display/redirect

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-175/2/display/redirect

liuxuezhao commented on src/common/debug.c:
```diff
@@ -35,103 +35,24 @@
 #define DAOS_DBG_MAX_LEN	(32)
 #define DAOS_FAC_MAX_LEN	(128)
 
+
```
just an extra blank line?

daosbuild1 commented:
Test stage Functional completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-175/2/display/redirect

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-175/3/display/redirect

daosbuild1 commented:
Test stage Functional completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-175/3/display/redirect

daosbuild1 commented:
Test stage Functional completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-175/4/display/redirect

gnailzenh commented:
It failed functional test?

jolivier23 commented:
I previously updated to earlier version of cart and it passed.  Then I updated to latest cart and hit the bug Alex mentioned in chat.   He has a patch for that.    I think I will update to that version.

But in this case, I've reverted cart version and think I'm just hitting the intermittent failure with pool create

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-175/5/display/redirect

brianjmurrell commented:
> But in this case, I've reverted cart version and think I'm just hitting the intermittent failure with pool create

The intermittent failure has been isolated to a single one of the test clusters and it has since been removed from the pool.  It would be worthwhile using "Restart from Stage" to rerun the Functional tests on this if there is any doubt.
